test/test_output_writers.jl

```diff
@@ -7,6 +7,58 @@ using Dates: Millisecond
 using Oceananigans: write_output!
 using Oceananigans.BoundaryConditions: PBC, FBC, ZFBC, ContinuousBoundaryFunction
 using Oceananigans.TimeSteppers: update_state!
+using Oceananigans.OutputWriters: construct_output
+
+
+function test_output_construction(model)
+
+    u, v, w = model.velocities
+    set!(model, u=(x, y, z)->y)
+
+    indices1 = (:, :, :)
+    indices2 = (1, 2:3, :)
+
+    # First we test fields
+    @test construct_output(u, model.grid, indices1, false) isa Field
+    @test construct_output(u, model.grid, indices2, false) isa Field
+
+    # Now we test windowed fields
+    u_sliced1 = Field(u, indices=indices1)
+    u_sliced2 = Field(u, indices=indices2)
+
+    @test construct_output(u_sliced1, model.grid, indices1, false) isa Field
+    @test construct_output(u_sliced1, model.grid, indices2, false) isa Field
+
+    @test construct_output(u_sliced2, model.grid, indices1, false) isa Field
+    @test construct_output(u_sliced2, model.grid, indices2, false) isa Field
+
+    @test parent(u_sliced2) != parent(construct_output(u_sliced1, model.grid, indices2, false)) # u_sliced2 should have halo regions
```

Comment:
Same for comparisons below.

Reply:
Isn't it more rigorous to compare the actual values though?

Reply:
I'm not sure how it could be. Testing field values is an indirect way of testing correct indices, that can pass even when `construct_output` fails catastrophically (ie because it copied data rather than reindexing). That flaw isn't really a big deal, I just think its harder to interpret the test when its indirect. We have to know what `parent` does to figure out what this test is doing, and that's non-trivial for a windowed field --- even subject to change, possibly...

Here we want to test that a windowed field was correctly reindexed in `construct_output`. We're probably 95% sure this works if (1) the indices are correct. But we could also test (2) that the underlying data points to the correct place.

Reply:
I mean it's more rigorous in this test case, when I know what positions have which values. I was trying to avoid mix-ups related to the different indexing you mentioned in array/subarray/offsetarray.

But I'll defer to you. I'll implement the `indices()` change.

Reply:
How is it possible to have two fields with the same indices but different values? I don't understand the mix up you're referring to (you mean, the error when calling `construct_output`?)

Reply:
> you mean, the error when calling `construct_output`?

Yes. Something like this: `ERROR: BoundsError: attempt to access 1×2×10 view(::Array{Float64, 3}, 4:4, 5:6, :) with eltype Float64 at index [4:4, 5:6, 4:7]`

I changed it to indices.